gulpfile.js/tests/configGenerator.test.js

```diff
@@ -0,0 +1,123 @@
+var fs = require('fs')
+var test = require('tape')
+var configGenerator = require('./../util/backstop/configGenerator')
+var deepClone = require('./../util/deepClone')
+var testFixturesPath = './gulpfile.js/tests/fixtures/configGenerator/'
+var config = {
+  compLib: {
+    port: 7000,
+    host: 'http://example',
+    baseDir: testFixturesPath
+  },
+  vrt: {
+    backstopConfigTemplate: testFixturesPath + 'backstop.template.json',
+    backstopConfig: testFixturesPath + 'backstop.json'
+  }
+}
+
+var removeFiles = function (files) {
+  return new Promise(function (resolve) {
+    files.forEach(function (file) {
+      fs.unlink(file, resolve)
+    })
+  })
+}
+
+var getBackstopConfigData = function () {
+  return new Promise(function (resolve, reject) {
+    fs.readFile(config.vrt.backstopConfig, function (err, data) {
```

Comment:
There's nothing wrong with using `require()` in a function. If anything you avoid having to handle 2 errors in the same callback (which looks more odd) and JSON parsing comes for free. if the file isn't there it'll just throw an error, so yes you'd still need to wrap it, but that's fine...

```
return new Promise(function (resolve, reject) {
	try {
	  resolve(require(config.vrt.backstopConfig))
	} catch (err) {
	  reject(err)
	}
})
```